### ISSUE TITLE
508-defect-3: Information alert and warning alert need alt text

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#54637]
## Point of contact

joshkimux

## Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

## Details

Both the information alert for the pact act and the warning alert for authorizing a doctor to release records do not have alt text despite what their visual design communicates.

https://user-images.githubusercontent.com/14154792/223226077-9b7feb53-ef82-404a-ad92-02ee3b485c53.mov

https://user-images.githubusercontent.com/14154792/223226194-7c512913-95ba-4a8c-bd29-721b690aab48.mov

## Reproduction steps

1. [Go to notice of evidence needed page](https://staging.va.gov/decision-reviews/supplemental-claim/file-supplemental-claim-form-20-0995/notice-of-evidence-needed) and using your screen reader confirm there is no alt text for the alert style
2. [On private medical records](https://staging.va.gov/decision-reviews/supplemental-claim/file-supplemental-claim-form-20-0995/supporting-evidence/private-medical-records-authorization), continue without checking the box and confirm no alt text is provided for the warning

### Notice of evidence
Given we are legally required to use the information alert, and the _visual design_ and icon of the alert communicates its importance, we should use "Information for" as alt text and update the close button to match.

- [ ] (**Information for**) If you have a presumptive condition `defect-3`
- [ ] "Close notification" button reads as "Close information alert" instead `defect-4`

Alt text likely will be easiest implemented through `sr-only` text within a `span`

### Private medical records
Given this is an _error_, [we should use the error alert](https://design.va.gov/components/alert#error-alert) as the user cannot proceed until they have done either task.

- [x] The error alert is used instead of the warning alert
- [ ] "Error" is read out by the screen reader as alt text (should come pre-packaged in the error alert component) `defect-3`

## References, articles, or WCAG support

1.
2.
3.
...

## Type of issue

- [X] Screenreader
- [ ] Keyboard
- [ ] Focus
- [ ] Headings
- [ ] Color
- [ ] Zoom
- [ ] Semantics
- [ ] Axe-core
- [ ] Something else

**Comment by Mottie** [comment]
For this, I can change the private medical evidence alert to an error type, but adding alt text would require a web component change - this confuses me, does this mean alt text on the close icon? Or adding an different `aria-label` on the close button?

**Comment by joshkimux** [comment]
Great question, I can clarify. Two things:
1. The information icon (i), would have alt text (likely through sr-only in a span) to read as “information for”— this is the heart of the ‘defect-level-2’
2. Since this is an information alert, and not a notification, we’d ideally want the close button to match up. So ideally, the close button’s aria-label would read as ‘close information alert’. That being said, this is a minor issue ‘defect-level-4’, if we need to switch the component, I don’t think it’s worth the effort. It may be better to write a DST issue ticket instead (I can do this!)

**Comment by Mottie** [comment]
@saderagsdale I think we can move this ticket into the product backlog once you've verified that this the notice of evidence alert is now an error alert.

**Comment by saderagsdale** [comment]
@eileen-coforma can you confirm?

**Comment by eileen-coforma** [comment]
@Mottie I think you mean the private-medical-records-authorization alert is an error. The notice of evidence needed alert (presumptive claims) should remain an informational alert.

@saderagsdale confirmed the private medical authorization alert is now an error.

**Comment by saderagsdale** [comment]
@joshkimux can you confirm if this needs to be fixed by the design system team?

**Comment by joshkimux** [comment]
Confirming it is now using the DST error alert which adds the `assertive` role and addresses the biggest issue in this ticket 🥳 

That being said, some minor issues remain for me to look into, notably-- DST alerts don't afford alt text. Will need to comb through documentation as to why that's the case. In the meantime, listing this as a `defect-3` pending `defect-4` depending on what I discover.

**Comment by joshkimux** [comment]
I've dug through documentation and [found this from a past a11y specialist](https://github.com/department-of-veterans-affairs/component-library/pull/137#discussion_r691275301) documenting the decision to remove `sr-only` text from the _warning_ error, reason being [the `alert` role would be repetitive aurally](https://github.com/department-of-veterans-affairs/component-library/pull/137#discussion_r691275301).

Need to dig into this a little more because this may not be consistent with how we handle error messages, which also use the `alert` role _and_ include "error" written within them.

**Comment by joshkimux** [comment]
Closing this out as this is more of a content consideration, and less of an accessibility one ✨ 

After talking to Brian, we agreed that:

> If the semantics of the alert aren't clear from the heading and the text of the alert, then that's a content/style guide problem.
